### PR TITLE
Updated examples and docs

### PR DESCRIPTION
**Pull request by robsdedude**
* Remove 3.5->4.x migration notes
 * Update examples to use `driver.execute_query` where appropriate
 * Updated links to external resources

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -47,99 +47,53 @@ Quick Example
...
-+ The ``neo4j://`` scheme replaces ``bolt+routing://`` and can be used for both clustered and single-instance configurations with Neo4j 4.0.
-
-
-
-See, https://neo4j.com/docs/migration-guide/4.0/upgrade-driver/#upgrade-driver-breakingchanges
 
+    def add_friend(driver, name, friend_name):
+        driver.execute_query(
+            "MERGE (a:Person {name: $name}) "
+            "MERGE (a)-[:KNOWS]->(friend:Person {name: $friend_name})",
```
Comment (fbiville) [inline README.rst]: maybe out of scope, but a better example would likely be:
```cypher
            "MERGE (a:Person {name: $name}) "
            "MERGE (friend:Person {name: $friend_name}) "
            "MERGE (a)-[:KNOWS]->(friend)"
```

Comment (fbiville) [reply README.rst]: the current impl would create duplicate friends if the KNOWS rel is missing which is rarely what you want to do

File: README.rst [hunk 2]
```diff
@@ -47,99 +47,53 @@ Quick Example
...
+    with GraphDatabase.driver(URI, auth=AUTH) as driver:
+        add_friend(driver, "Arthur", "Guinevere")
+        add_friend(driver, "Arthur", "Lancelot")
+        add_friend(driver, "Arthur", "Merlin")
+        print_friends(driver, "Arthur")
 
 
-Other Information
-=================
+Further Information
```
Comment (fbiville) [inline README.rst:83]: maybe add GraphAcademy as well?

File: docs/source/index.rst [hunk 3]
```diff
@@ -100,60 +108,39 @@ To deactivate the current active virtual environment, use:
...
     URI = "neo4j://localhost:7687"
     AUTH = ("neo4j", "password")
 
 
-    def create_person(tx, name):
-        tx.run("CREATE (a:Person {name: $name})", name=name)
+    def add_friend(driver, name, friend_name):
+        driver.execute_query(
+            "MERGE (a:Person {name: $name}) "
+            "MERGE (a)-[:KNOWS]->(friend:Person {name: $friend_name})",
```
Comment (fbiville) [inline docs/source/index.rst]: same as previous comment

File: docs/source/index.rst [hunk 4]
```diff
@@ -163,99 +150,110 @@ Example Application
...
             app.find_person("Alice")
         finally:
             app.close()
 
 
-*****************
-Other Information
-*****************
+*******************
+Further Information
```
Comment (fbiville) [inline docs/source/index.rst:243]: same as above

File: src/neo4j/_async/auth_management.py [hunk 5]
```diff
@@ -159,9 +159,11 @@ def expiration_based(
             The provider function **must not** interact with the driver in any
             way as this can cause deadlocks and undefined behaviour.
 
-            The provider function only ever return auth information belonging
-            to the same identity.
+            The provider must function only ever return auth information
```
Comment (fbiville) [inline src/neo4j/_async/auth_management.py]: typo?

File: src/neo4j/_sync/auth_management.py [hunk 6]
```diff
@@ -159,9 +159,11 @@ def expiration_based(
             The provider function **must not** interact with the driver in any
             way as this can cause deadlocks and undefined behaviour.
 
-            The provider function only ever return auth information belonging
-            to the same identity.
+            The provider must function only ever return auth information
```
Comment (fbiville) [inline src/neo4j/_sync/auth_management.py]: typo?

File: tests/integration/test_readme.py [hunk 7]
```diff
@@ -16,55 +16,56 @@
...
+    AUTH = ("neo4j", "password")
+
 
-    def print_friends(tx, name):
-        query = ("MATCH (a:Person)-[:KNOWS]->(friend) WHERE a.name = $name "
-                 "RETURN friend.name ORDER BY friend.name")
-        for record in tx.run(query, name=name):
+    def add_friend(driver, name, friend_name):
+        driver.execute_query(
+            "MERGE (a:Person {name: $name}) "
```
Comment (fbiville) [inline tests/integration/test_readme.py:40]: same as before

**Review by fbiville — CHANGES_REQUESTED** [review]

**Review by fbiville — APPROVED** [review]

**Review by AndyHeap-NeoTech — APPROVED** [review]
All good.

**Pull request merged**